deploy/modules/gcp/tidb-operator/variables.tf

```diff
@@ -11,14 +11,20 @@ variable "subnetwork_name" {
   description = "The name of the subnetwork in which to place the cluster"
 }
 
-variable "gcp_region" {
-  description = "The GCP region"
-}
-
 variable "gcp_project" {
   description = "The GCP project name"
 }
 
+variable "location" {
+  description = "The GKE cluster location. If you specify a zone (such as us-central1-a), the cluster will be a zonal cluster with a single cluster master. If you specify a region (such as us-west1), the cluster will be a regional cluster with multiple masters spread across zones in the region."
+  type        = string
+}
+
+variable "node_locations" {
+  description = "The list of zones in which the cluster's nodes should be located. These must be in the same region as the cluster zone for zonal clusters, or in the region of a regional cluster. In a multi-zonal cluster, the number of nodes specified in initial_node_count is created in all specified zones as well as the primary zone. If specified for a regional cluster, nodes will be created in only these zones."
```

Comment:
What is the purpose of this for a single-zone cluster? Doesn't the above location variable already specify this properly?

Reply:
oh, I see, there are zonal clusters that are multi-zone clusters. I wonder why not just use the regional cluster though?

Reply:
yes, if you need to run a multi-zone cluster, it's better to use a regional cluster. However, if users want a small cluster for testing or for low network latency between work nodes, they can use a zonal cluster and run nodes in a single zone. 

`node_locations` can also be used to choose node locations for the regional cluster too (at least two locations for a regional cluster).

Currently, in GCP, some types of machines are only available in a few zones, e.g. in region `us-central1` [c2-standard machines](https://cloud.google.com/compute/docs/machine-types#c2_machine_types) are only available in `us-central1-b` and `us-central1-c` zones.

If users want to use these compute-optimized machines, they must specify `node_locations` to deploy nodes in these zones.